Per-org VAN settings redux

jeffm2001 opened the pull request:
# Fixes #1693 

## Description

This is a rework of #1813, implementing a per-organization VAN credential interface as a service manager. A few notes:
- I added a `setFeatures` method to cacheable organization. This mirrors the setFeatures method available on campaigns, and should make it easier for future extensions to store organization config without duplicating a bunch of code.
- Decrypting keys stored in the database is a bit trickier than it was in #1813, because the secret manager is now `async`, but `getConfig` is not. Potentially, we should make getConfig async, but I thought that was too far to go in this PR.

![CleanShot 2022-08-01 at 13 09 13](https://user-images.githubusercontent.com/18371709/182204731-c1eb8bac-666a-42a4-ba31-f982530597f0.png)

# Checklist:

- [x] I have manually tested my changes on desktop and mobile
- [x] The test suite passes locally with my changes
- [x] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [x] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#submitting-your-pull-request), or has a documented reason in the description why it’s longer
- [x] I have made any necessary changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] My PR is labeled [WIP] if it is in progress

Arique1104 reviewed (changes requested):
Hey @jeffm2001, I am trying to get the NGPVAN Integration UI section in settings.  Similar to how you demonstrated on this pull request.  I'm trying to figure out what step I'm missing to test the integration on my end.

I've been following the [REFERENCE-NGPVAN User Guide](https://github.com/aflcio/Spoke/blob/1693-multi-van-service-manager/docs/REFERENCE-NGPVAN_user_guide.md)

Maybe there's a step missing that isn't apparent to me?

- `NGP_VAN_WEBHOOK_BASE_URL` is set to same value as `BASE_URL`
- Not sure which way to go with the `appending |1 after API Key` so I left it out.
- Set the `NGP_VAN_API_KEY` 
- and the `NGP_VAN_APP_NAME` env variables
- Wasn't sure how to do the Redis `FLUSHDB` so I pressed the `clear cache` button in `settings`
- Wasn't sure which file we fix the column for the organization table (and was wondering if that was done in this pull request?) I skipped this step
- Set the `CONTACT_LOADER` to `ngpvan`
- This is where I learned that I could load a saved list (not a saved search) and I remember I created some test lists in my dev environment while working on this issue with them. [Click this link to check it out.](https://github.com/MoveOnOrg/Spoke/pull/2122)
- Set `ACTION_HANDLER` to `ngpvan`
- Set `MESSAGEE_HANDLER` to `ngpvan`
- However I was lost with changing text canvass result in environment variables. `DEFAULT_NGP_VAN_INITIAL_TEXT_CANVASS_RESULT` changes the contact result from "Texted" to anything we'd like.  So I left that alone.

Is there a step I'm missing to get this to show up right on my computer?

jeffm2001 commented:
@Arique1104 I think the step missing was enabling the new service manager extension. I just added it to documentation.

I removed the previous text about directly setting the variables in the database because, while it's still technically possible to do it that way, storing unencrypted API credentials in a database is generally considered a bad practice.

I also updated the documentation about using MyCampaign, since it was outdated.

Arique1104 commented on src/extensions/service-managers/ngpvan/index.js:
```diff
@@ -0,0 +1,61 @@
+    data: {
+      NGP_VAN_API_KEY_ENCRYPTED: NGP_VAN_API_KEY_ENCRYPTED && "<Encrypted>",
+      NGP_VAN_APP_NAME,
+      NGP_VAN_DATABASE_MODE: String(NGP_VAN_DATABASE_MODE)
```
This edit allowed for the application to present the expected UI in the organization's setting's page.  Thank you for your work Jeff, I'm excited to provide this to the community!

Arique1104 reviewed (approved):
This feature is ready for the community to use.
